stl/src/fiopen.cpp

```diff
@@ -58,13 +58,10 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
 
     mode &= ~(ios_base::ate | ios_base::_Nocreate | ios_base::_Noreplace);
 
-    int n = 0;
-    while (valid[n] != 0 && valid[n] != mode) { // look for a valid mode
-        ++n;
-    }
-
-    if (valid[n] == 0) {
-        return nullptr; // no valid mode
+    for (unsigned int n = 0; valid[n] != mode; ++n) { // look for a valid mode
+        if (valid[n] == 0) {

```

Comment:
If `mode == 0`, this function previously returned `nullptr` here but now continues processing.

Reply:
in that case, let's save time by checking

Reply:
mode being 0 should be the first thing we check, not the last in that case, since there is no avoiding this check anyway